packages/strapi-plugin-upload/services/Upload.js

```diff
@@ -43,10 +43,9 @@ const combineFilters = params => {
 
 module.exports = {
   formatFileInfo({ filename, type, size }, fileInfo = {}, metas = {}) {
-    const ext = '.' + mime.extension(type) || path.extname(filename);
-    const baseName = path.basename(filename, path.extname(filename));
+    const ext = path.extname(filename);
```

Comment:
This doesn't work in the case you upload a file without an extension or with an invalid extension.

Reply:
It works without extension - just checked. It won't show thumbnail but will upload the file and make it downloadable.

Reply:
So it doesn't match the behavior we are aiming for then. We should at least add the extension if the file doesn't have any. And what do you suggest to do in the case we have a file with a random . in the name ?

Reply:
IMO, a user chooses how he wants to keep his media library. Usually, there is a folder with images I need to upload.
If the server will change something in name (even extension) it will cause incompatibility between the source and library after upload. So, I think it should be consistent.

Even if a file has few dots in its name - it should be ok from my perspective. It needs to leave file name untouched. It will let the user more control and less mistakes.

Reply:
We had do to this to handled  bugs we saw. I can't remember well which one though. I think it had to do with uploading a file with another format extension like uploading a png but naming it with .svg or sth... 
Will need to do more testing before we change this. 

For the naming part We could use the full filename as it is only a display name that's not really problematic

Reply:
I just checked to upload jpg file as png - everything went well and it did not have any impact.
It is still shown in library (as png, of course) but works great.

Reply:
As I told you this doesn't work with svg. Just reverting to extension name without doing more logic is out of the question as of now. You can check other media lib (e.g cloudinary) and you will see this is a pretty commong behavior to rename the extension so it matches the mime-type. 

What would be a good compromise is to check if the extension provided corresponds to the mime type and leave it as is so you would keep the original extension name and otherwise use the correct one so the file can be displayed as much as possible. 

For the name we can keep the final computed extension value in it but it needs to be removed in the hash so the url make more sense.

Reply:
I see you did not get me correct.
I do not try to fix mime-type to be set correct. I just let to file be named as user wish.
The code doesn't change extension according to different mime-type.
It CAN'T detect correct mime-type right now if you don't put correct file extension!
So, our discussion is out of patch scope.

What now happens: if I try to upload img.jpg then strapi converts file name to img.jpeg.
Only this. If I rename it to svg then it will not change it to svg because mime-type detection is based on file extension and not on file content.

If you want to open new PR for it - please, let me know but it is not related to this one.

So, the only thing I do with this - to keep file name original.

Thanks.

Reply:
Thanks for the clarification. 
Will need to test specific usecase to make sure this doesn't break certain edges cases.

Reply:
Is there a work-around? I need to upload .gltf, .glb and .usdz files. But they are all turned into .bin which I cant use anymore.

Reply:
I would say, if the mime-type detection works, use that.
Otherwise, preserve the file extension.
There is no issue with multiple dots. you just do `const parts  =  filename.split('.')` and pick `parts[parts.length-1]`
It is, IMHO, the best of both worlds

Reply:
I have to repeat:
Mime-type detection is based on file extension and not on file content.
So, it is useless: .png stays .png, but .jpg gets .jpeg which is the worst for the user.
if mime-type detection would based on file content even then the file extension has to stay untouched - don't forget - it is an user input! We shouln't change a content for him. As CMS we store user content (file name is a part of it) but we dont have permissions to change it, IMHO.

Reply:
the mime-type is useful for setting the content-type header when you're serving the file
otherwise, I concur, you should preserver whatever the user input/data you got